src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseLongTermRetentionBackup.cs

```diff
@@ -0,0 +1,216 @@
+                ServerName,
+                DatabaseName,
+                BackupName,
+                ResourceGroupName,
+                null,
+                null);
+        }
+
+        /// <summary>
+        /// User input to apply to model
+        /// </summary>
+        /// <param name="model">Model retrieved from service</param>
+        /// <returns>The model that was passed in</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> ApplyUserInputToModel(
+            IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> model)
+        {
+            model.First().RequestedBackupStorageRedundancy = BackupStorageRedundancy;
+            return model;
+        }
+
+        /// <summary>
+        /// Persist changes to ser
+        /// </summary>
+        /// <param name="entity">The output of apply user input to model</param>
+        /// <returns>The input entity</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> PersistChanges(
+            IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> entity)
+        {
+            if (ShouldProcess(DatabaseName))
+            {
+                return new List<AzureSqlDatabaseLongTermRetentionBackupModel>()
+                {
+                    ModelAdapter.SetDatabaseLongTermRetentionBackup(entity.First())
+                };
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Entry point for the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            if (!string.IsNullOrWhiteSpace(ResourceId))
+            {
+                ParseLongTermRentionBackupResourceId(ResourceId);
+            }
+
+            if (ShouldProcess(this.BackupName))
+            {
+                base.ExecuteCmdlet();
+            }
+        }
+
+        /// <summary>
+        /// Parse the longTermRetentionBackup resource Id
+        /// </summary>
+        /// <param name="resourceId"></param>
+        private void ParseLongTermRentionBackupResourceId(string resourceId)
+        {
+            int offset = 0;
+            string[] tokens = resourceId.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+            if (tokens.Length == 14 || tokens.Length == 12)
+            {
+                if (tokens.Length == 14)
+                {
+                    ResourceGroupName = tokens[3];
+                    offset = 2;
+                }
+                else
+                {
+                    ResourceGroupName = null;
+                }
+
+                Location = tokens[5 + offset];
+                ServerName = tokens[7 + offset];
+                DatabaseName = tokens[9 + offset];
+                BackupName = tokens[11 + offset];
+            }
+            else
+            {
+                throw new ArgumentException("Invalid parameter", "ResourceId");
```

Comment:
It's suggested to throw AzPSArgumentException or any AzPs*Exception suitable for this case as far as possible. See https://github.com/Azure/azure-powershell-common/tree/master/src/Common/Exceptions